anomalib/models/components/base/anomaly_module.py

```diff
@@ -50,78 +53,81 @@ def __init__(self):
         self.image_metrics: AnomalibMetricCollection
         self.pixel_metrics: AnomalibMetricCollection
 
-    def forward(self, batch):  # pylint: disable=arguments-differ
+    def forward(self, batch: dict[str, str | Tensor], *args, **kwargs) -> Any:
         """Forward-pass input tensor to the module.
 
         Args:
-            batch (Tensor): Input Tensor
+            batch (dict[str, str | Tensor]): Input batch.
 
         Returns:
             Tensor: Output tensor from the model.
         """
         return self.model(batch)
 
-    def validation_step(self, batch, batch_idx) -> dict:  # type: ignore  # pylint: disable=arguments-differ
+    def validation_step(self, batch: dict[str, str | Tensor], *args, **kwargs) -> STEP_OUTPUT:
         """To be implemented in the subclasses."""
         raise NotImplementedError
 
-    def predict_step(self, batch: Any, batch_idx: int, _dataloader_idx: Optional[int] = None) -> Any:
+    def predict_step(self, batch: Any, batch_idx: int, dataloader_idx: int = 0) -> Any:
         """Step function called during :meth:`~pytorch_lightning.trainer.trainer.Trainer.predict`.
 
         By default, it calls :meth:`~pytorch_lightning.core.lightning.LightningModule.forward`.
         Override to add any processing logic.
 
         Args:
-            batch (Tensor): Current batch
+            batch (Any): Current batch
             batch_idx (int): Index of current batch
-            _dataloader_idx (int): Index of the current dataloader
+            dataloader_idx (int): Index of the current dataloader
 
         Return:
             Predicted output
         """
-        outputs = self.validation_step(batch, batch_idx)
+        del batch_idx, dataloader_idx  # These variables are not used.
```

Comment:
can't we solve this with `_batch_idx` and `_dataloader_idx`

Reply:
Again this is based on this style guide, but if we agree to use _batch_idx, then I'm fine. Of course we need to ensure if pylint is happy.
https://google.github.io/styleguide/pyguide.html#214-decision

Reply:
The guideline states the following though
> Other common forms of suppressing this warning include using ‘_’ as the identifier for the unused argument or prefixing the argument name with ‘unused_’, or assigning them to ‘_’. These forms are allowed but no longer encouraged. These break callers that pass arguments by name and do not enforce that the arguments are actually unused.